packages/cli/test/unit/util/dev/parse-query-string.test.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  parseQueryString,
+  formatQueryString,
+} from '../../../../src/util/dev/parse-query-string';
+
+describe('parseQueryString', () => {
+  it('should parse to Map and format back to original String', async () => {
+    const querystring = '?a&b=1&c=2&c=3&d=&space%20bar=4&html=%3Ch1%3E';
+    const parsed = parseQueryString(querystring);
+    const map = new Map<string, string[]>([
+      ['a', []],
+      ['b', ['1']],
+      ['c', ['2', '3']],
+      ['d', ['']],
+      ['space bar', ['4']],
+      ['html', ['<h1>']],
+    ]);
+    expect(parsed).toEqual(map);
+    const format = formatQueryString(parsed);
+    expect(format).toEqual(querystring);
+  });
+  it('should work with empty string', async () => {
+    const parsed = parseQueryString('');
+    expect(parsed).toEqual(new Map());
+    const format = formatQueryString(parsed);
+    expect(format).toEqual(undefined);
+  });
+  it('should work with question mark', async () => {
+    const parsed = parseQueryString('?');
+    expect(parsed).toEqual(new Map());
+    const format = formatQueryString(parsed);
+    expect(format).toEqual(undefined);
+  });
+  it('should work without question mark', async () => {
```

Comment:
What is this case for?

Reply:
These are just unit tests so I wanted to have full coverage. I don't think this is actually possible when passing the result of `url.parse(foo).search` into this function.

Although now that I look, this seems questionable: https://github.com/vercel/vercel/blob/e8901e7be783d673863e2154c40fa084bd4c6902/packages/cli/src/util/dev/server.ts#L1543